src/main/java/org/kuali/kfs/kim/impl/role/RoleServiceImpl.java

```diff
@@ -346,8 +341,9 @@ protected void putRoleInCache(RoleLite role) {
         return new ArrayList<>(roleMap.values());
     }
 
-    // bacport redis annotation
-    @Cacheable(cacheNames = Role.CACHE_NAME, key = "'{getRoleByNamespaceCodeAndName}-namespaceCode=' + #p0 + '|' + 'name=' + #p1")
+    // bacport redis annotation; backport fix on FINP-8169
+    @Cacheable(cacheNames = Role.CACHE_NAME,
+            key = "'{" + Role.CACHE_NAME + "}-namespaceCode=' + #p0 + '|' + 'name='+ #p1")
```

Comment:
The KualiCo fix leaves the hyphen in this cache key, even though the related namespace-and-name keys in the other methods above don't have this hyphen. The cache will still work properly, it's just that this method's cache entries won't be interchangeable with those from the relevant get/put methods farther up.

If KualiCo is still working on the FINP-8210 follow-up fixes, we could see if they could include changing this key as part of that work.